pkg/chain/ethereum/connect.go

```diff
@@ -93,7 +94,9 @@ func connectWithClient(
 		transactionMutex: &sync.Mutex{},
 	}
 
-	blockCounter, err := blockcounter.CreateBlockCounter(pv.client)
+	blockCounter, err := ethlike.CreateBlockCounter(
+		ethutil.NewBlockSourceAdapter(pv.client),
+	)
```

Comment:
Can we have this logic hidden inside of `keep-common`s `ethutil`? I don't have the full context grasped yet but I think it could make sense to have `ethereum` talking only to `ethutil` and `ethutil` using some components from `ethlike` internally.

Reply:
Done. Please see:
- https://github.com/keep-network/keep-common/pull/64/commits/a8c0a5ca0a5d094451e6d66d2d5d4c3130c77bce
- https://github.com/keep-network/keep-core/pull/2322/commits/07bf236ec33bb6f8f75bd3a17c483df182395343